src/agni/client/action/UserActionHandler.java

```diff
@@ -1,16 +1,52 @@
 package agni.client.action;
 
+import java.util.Arrays;
+
 import agni.client.communication.MessageSender;
 
 public class UserActionHandler {
     private MessageSender messageSender;
+    final private int HEADER_LENGTH_SIZE = 4;
+    final private int MESSAGE_TYPE_SIZE = 1;
+    final private int USER_ACTION_TYPE_LENGTH = 1;
+    final private int EXTRA_ARGUMENT_SIZE = 1;
+    final private byte MESSAGE_TYPE = 0x04;
 
     public UserActionHandler(MessageSender messageSender) {
         this.messageSender = messageSender;
     }
 
-    public void requestUserAction(byte userActionType) {
-
+    public void requestUserAction(byte userActionType, byte extraArg, boolean extraArgUsed) {
```

Comment:
the extraArg should be type string. The views shouldn't need to worry about converting a string to a byte.